### ISSUE TITLE
LibreOffice Calc: report selected cell coordinates

### ISSUE THREAD

**Issue by surfer0627** [nvaccess/nvda#6897]
This is a feature request that NVDA report selected cell  coordinates.
Steps to reproduce:
1. Open Calc.
2. Press shift+downArrow.
NVDA report: nothing
expected: a1 through a2 selected
3. press ctrl+home to move to top.
4. Press shift+space to select entire row
NVDA report: space
expected: a1 through amj1 selected

**Comment by zahra21** [comment]
hi.
i confirm this issue.
does nvda support reporting of selection in microsoft excel?
if you found solution for this, please inform me.
thanks and God bless all.

**Comment by leonardder** [comment]
LibreOffice Table Cells don't implement IAccessibleTableCell, and therefore there's no way to get the row and column span. LibreOffice itself also does not expose the merged state in the accessible name of the cell.

**Comment by Adriani90** [comment]
@zahra21 yes in Excel this is supported. In Libre Calc this is still an issue. Right?

**Comment by surfer0627** [comment]
It seems that LibreOffice developer Tamás Zolnai committed a patch related to this issue (adding selection-changed event).

[bug 93825 – Missing accessibility events when selecting/deselecting cells in Calc](https://bugs.documentfoundation.org/show_bug.cgi?id=93825)

cc @leonardder: Could you please check, if this is helpful?

Thanks.

**Comment by leonardder** [comment]
Is this in a release already?

**Comment by surfer0627** [comment]
Yes, developer Tamás Zolnai pointed out that it is available in 5.2.4. leonardder

**Comment by leonardder** [comment]
I tested with the most recent version, but without luck I'm afraid.

**Comment by surfer0627** [comment]
In this case, LibreOffice developer said "selection change event is added."
But, this patch does not fix screen reader NVDA "report selected cell coordinates".
What should we do (Let LibreOffice developer know the gap)?
Or how to use this **"selection change event** to implement feature "report selected cell coordinates?

**Comment by leonardder** [comment]
I actually have the feeling that, though there is now a selection change 
event, there is no focus event fired when selecting another cell. We 
might be able to fix this on our end, but I'll have to investigate this. 
I'd rather have LibreOffice doing this correctly, but for me personally, 
it is not that much of a high priority honestly.

**Comment by nktech1135** [comment]
Is anyone working on this? It's still an issue in libreoffice 7.2 and nvda 2020.3
A fix would be greatly appreciated.

**Comment by leonardder** [comment]
Apart from the patch refered to in https://github.com/nvaccess/nvda/issues/6897#issuecomment-519355077, it looks like LibreOffice people aren't that accessibility minded unfortunately.

**Comment by nktech1135** [comment]
> it looks like LibreOffice people aren't that accessibility minded unfortunately.

Do we have examples of this? Were the correct bugs ever filed against libreoffice to make them aware of the issue? I'm not a developer so can't file the bugs myself as i wouldn't know what to point at or what the behind the seens tech speak would be.

**Comment by feerrenrut** [comment]
A number of accessibility bugs filed, many [issues here with the `app/libreoffice` label](https://github.com/nvaccess/nvda/labels/app%2Flibreoffice) have links to bugs on the issue tracker for libreoffice. 

I have tried to reach out to a few developers who work on libreoffice, for the record NV Access is willing to work with libreoffice developers to help solve these issues. They can contact us via info@nvaccess.org to start a conversation.

**Comment by Qchristensen** [comment]
There is an issue on LibreOfice's bug tracker for this, which goes back to 2016: https://bugs.documentfoundation.org/show_bug.cgi?id=100086 (with references to an older issue again).
